includes/enrolment/class-sensei-course-enrolment.php

```diff
@@ -444,4 +445,87 @@ public function reset_course_enrolment_salt() {
 
 		return $new_salt;
 	}
+
+	/**
+	 * Remove learner from the course, overriding the providers rule.
+	 *
+	 * @param int $user_id User ID.
+	 *
+	 * @return boolean Success flag.
+	 */
+	public function remove_learner( $user_id ) {
+		$removed_learners = $this->get_removed_learners();
+
+		if ( isset( $removed_learners[ $user_id ] ) ) {
+			return false;
+		}
+
+		$removed_learners[ $user_id ] = [ 'date' => time() ];
+
+		$this->save_enrolment( $user_id, false );
+
+		return $this->update_removed_learners( $removed_learners );
+	}
+
+	/**
+	 * Restore removed learner enrolment, giving the control back to the providers.
+	 *
+	 * @param int $user_id User ID.
+	 *
+	 * @return boolean Success flag.
+	 */
+	public function restore_removed_learner( $user_id ) {
+		$removed_learners = $this->get_removed_learners();
+
+		unset( $removed_learners[ $user_id ] );
+
+		return $this->update_removed_learners( $removed_learners );
+	}
+
+	/**
+	 * Check if the user is removed.
+	 *
+	 * @param int $user_id User ID.
+	 *
+	 * @return boolean Whether the learner is removed.
+	 */
+	public function check_removed_learner( $user_id ) {
+		$removed_learners = $this->get_removed_learners();
+
+		return in_array( $user_id, array_keys( $removed_learners ), true );
```

Comment:
`array_key_exists` is simpler.

Reply:
Fixed in 582293f8e5e58b31dd0b46f041b8f85f8d703a14

Reply:
TIL 😊